target/linux/ath79/dts/ar7100_mikrotik_routerboard-4xx.dtsi

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: GPL-1.0-or-later OR MIT
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar7100.dtsi"
+
+/ {
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	aliases {
+		led-boot = &led_user;
+		led-failsafe = &led_user;
+		led-running = &led_user;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_user: user {
+			label = "mikrotik:yellow:user";
+			gpios = <&gpio 4 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		led1 {
+			label = "mikrotik:green:led1";
+			gpios = <&cpld 0 GPIO_ACTIVE_LOW>;
+		};
+
+		led2 {
+			label = "mikrotik:green:led2";
+			gpios = <&cpld 2 GPIO_ACTIVE_LOW>;
+		};
+
+		led3 {
+			label = "mikrotik:green:led3";
+			gpios = <&cpld 1 GPIO_ACTIVE_LOW>;
+		};
+
+		led4 {
+			label = "mikrotik:green:led4";
+			gpios = <&cpld 3 GPIO_ACTIVE_LOW>;
+		};
+
+		led5 {
+			label = "mikrotik:green:led5";
+			gpios = <&cpld 8 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	nand {
+		compatible = "mikrotik,rb4xx-nand";
+
+		gpios = <&cpld 5 GPIO_ACTIVE_HIGH>, // ALE
+			<&cpld 6 GPIO_ACTIVE_HIGH>, // CLE
+			<&cpld 7 GPIO_ACTIVE_HIGH>, // nCE
+			<&gpio 5 GPIO_ACTIVE_HIGH>; // RDY
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "booter";
+				reg = <0x0000000 0x0040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "kernel";
+				reg = <0x0040000 0x03c0000>;
+			};
+
+			partition@400000 {
+				label = "ubi";
+				reg = <0x0400000 0x7c00000>;
+			};
+		};
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&gpio {
+	ngpios = <31>;
+};
+
+&pcie0 {
+	// NOT TESTED (n.b. ath79_pci_irq struct has the slot/pin/IRQ mappings if needed)
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	compatible = "mikrotik,rb4xx-spi";
+
+	flash@0 {
+		compatible = "pm25lv512", "jedec,spi-nor";
```

Comment:
Is the pm25lv512 compatible needed for detection?

Reply:
Just tested, and not it is not needed with this newer drivers. Removed it.

Reply:
I'm going to take back that previous comment as the following error message is shown without it:

`m25p80 spi0.0: unrecognized JEDEC id bytes: ff, ff, ff`